### DISCUSSION TITLE
How can I run a binary with environment variable from `/etc/environment`

### DISCUSSION THREAD

**Discussion by rew1nter** [davatorium/rofi#2007] [Q&A]
I have an environment variable set in `/etc/environment` that my executable at `/usr/local/bin/` looks for. It works perfectly fine when started from the terminal but doesn't work when started from rofi. Am I missing something?

**Comment by DaveDavenport** [comment]
rofi inherits the environment it is launched from,  it passes this environment to the program it launches.
So if the environment is correct where rofi is launched from, it should work.

**Comment by lbonn** [comment]
@rew1nter this can be relevant https://superuser.com/questions/664169/what-is-the-difference-between-etc-environment-and-etc-profile

Usually `~/.profile` or `~/.bash_profile` etc is where you want to set your variables.

If it's really a multi-user variable, distros usually have a `/etc/profile.d` where you can drop a .sh file

**Comment by rew1nter** [comment]
Initially I tried bashrc but it wasn't working either. 

I just launch rofi from sxhkd with a keybind. Is there way to control the environment of sxhkd?

**Reply by lbonn** [reply]
Well yes but it's not really a rofi question but a general linux issue at this point. It will depend a bit on your system and how the environment is read but basically you should look for a way to set your environment so that your WM inherits from it.

If I had to guess actually, this might help https://wiki.archlinux.org/title/Xprofile. Otherwise try to find help related to your distro and display manager (eg lightdm...) if you use one.
